### PR TITLE
Fix fd_sync Error for Preopened Base Directory

### PR DESCRIPTION
**Pull request by MeenuyD**
Fix `fdSync` to Handle Directories in Linux Implementation.
I have updated the `fdSync` function in `inode-linux.cpp` to correctly handle synchronization requests for directories. The function now checks if the file descriptor is a directory and treats it as syncable, in addition to performing the `fsync` operation for regular files. This change ensures compatibility with other runtimes and aligns the behavior with the WASI specification.
#3039

### CODE REVIEW

**Comment by juntao** [comment]
Hello, I am a [code review agent](https://github.com/flows-network/github-pr-review/) on [flows.network](https://flows.network/).

It could take a few minutes for me to analyze this PR. Relax, grab a cup of coffee and check back later. Thanks!

**Review by hydai — CHANGES_REQUESTED** [review]
The coding style is not compatible with our standard, please fix it.

**Comment by MeenuyD** [comment]
> The coding style is not compatible with our standard, please fix it.

How should I fix this to compatible with standard

**Comment by MeenuyD** [comment]
Could you please help me in this

File: lib/host/wasi/inode-linux.cpp [hunk 1]
```diff
@@ -549,6 +549,10 @@ WasiExpect<void> INode::fdSeek(__wasi_filedelta_t Offset,
 }
 
 WasiExpect<void> INode::fdSync() const noexcept {
+  struct stat fdStat;
+  if (fstat(Fd, &fdStat) == 0 && S_ISDIR(fdStat.st_mode)) {
+    return {};
+  }
```
Comment (dannypsnl) [inline lib/host/wasi/inode-linux.cpp:555]: This logic is: if we get a directory, we ignore it. I don't think this is expected behaviour.

**Review by dannypsnl — CHANGES_REQUESTED** [review]

**Comment by hydai** [comment]
Closing due to inactive.

**Pull request closed without merging**